### PR TITLE
registry

### PR DESCRIPTION
**Pull request by realcodywburns**
from issue #5: a token registry.
functions:
issue #39 
`joinTheReg()`: adds contract to registry. costs a price (1 etc currently, set by admins)

for issue #61 :
`regQuick()`; get any tokens short info [address, name, ticker, icon] quicker than full array
`regData()`: get a tokens long info

For tokens:
`tAddr`: tokens address,
`tSale`: the crowd sale address,
`tName`: the tokens long name,
`tSymbol`: tiker symbol,
`tDecimal`: how many decimal places,
`tType`: erc20 or 23 or whatever(bec);
`tIcon`: symbol for the token(string so either emoji or img url,
`tURL`: the owners website or whitepaper link,
`tBlerb`: short description of the token.

issue #47 : 
`count()` : running count of contracts on registry
event `newToken` triggered when a new token is added to the registry

registry deployed at:
0xf4d139fE4f73725dcc6eb364e4593a00F07DC0a9

Current price is `1 ETC` to list a token using joinTheReg.

BEC is stored in position `0`

adminOnly functions that are not listed in contract.js:  
modCategory() : 
`0` add a token with out paying, requires all the fields to be filled out
`1` delete a token at index number
`3` change the price to list
`4` rename the registry

modAdmin():
`0` add new admin
`1` delete admin
`3` change admin info

### CODE REVIEW

File: src/lib/contract.js [hunk 1]
```diff
@@ -92,3 +95,45 @@ export const ERC20Funcs = [
...
+          {name:"_tURL",type:"string"},
+          {name:"_tBlerb",type:"string"}]
+});
+
+export const regCount =  Immutable.fromJS({
+        name: "Count",
+        inputs:[],
+        outputs:[{ name: '', type: 'uint'}]
+      });
+
```
Comment (elaineo) [inline src/lib/contract.js]: ok, so proper usage is to call `regCount`, then iterate through the indices with `getArray`?

File: contracts/manReg.sol [hunk 2]
```diff
@@ -200,20 +204,24 @@ function ownerCheck() constant returns(address){
   return owner;
 }
 
-function getArray(uint _index) constant returns (address,address,string,string,uint,string,string)  // NOTE 3 see below
+function getArray(uint _index) constant returns (address,address,string,string,uint,string,string,string,string)  // NOTE 3 see below
     {
         token t = tokens[_index];
-    	return (t.tAddr,t.tSale,t.tName,t.tSymbol, t.tDecimal, t.tType,t.tIcon);
+    	return (t.tAddr,t.tSale,t.tName,t.tSymbol, t.tDecimal, t.tType,t.tIcon, t.tURL,t.tBlerb);
```
Comment (elaineo) [inline contracts/manReg.sol]: why not return the struct?

File: src/lib/contract.js (hunk 1 above)
Comment (realcodywburns) [inline src/lib/contract.js]: yes that is how I have been using it. or iterate through the index until the address is `null`

**Comment by realcodywburns** [comment]
Since it is a public, we could add tokenmint generated tokens for free using `modCategory` add and the public can register their own tokens with the `register()` for a fee

File: contracts/manReg.sol (hunk 2 above)
Comment (realcodywburns) [inline contracts/manReg.sol]: solidity doesn't handle passing structs to the abi

**Comment by elaineo** [comment]
how do we add tokenmint-generated tokens? you mean manually with an admin account?

**Comment by realcodywburns** [comment]
I could add icomachine, or another contract, as an admin and then it can use the extra features

**Comment by elaineo** [comment]
Probably best to do this from a contract to avoid saving an account on the tokenmint server. I can work on a new version of ICOmachine that also adds to registry

**Pull request merged**